pinot-core/src/main/java/org/apache/pinot/core/operator/combine/merger/TimeSeriesAggResultsBlockMerger.java

```diff
@@ -44,6 +49,14 @@ public void mergeResultsBlocks(TimeSeriesResultsBlock mergedBlock, TimeSeriesRes
       BaseTimeSeriesBuilder newTimeSeriesToMerge = entry.getValue();
       if (currentTimeSeriesBuilder == null) {
         currentTimeSeriesBlock.getSeriesBuilderMap().put(seriesHash, newTimeSeriesToMerge);
+        final long currentUniqueSeries = currentTimeSeriesBlock.getSeriesBuilderMap().size();
+        final long numBuckets = currentTimeSeriesBlock.getTimeBuckets().getNumBuckets();
+        Preconditions.checkState(currentUniqueSeries * numBuckets <= _maxDataPointsLimit,
```

Comment:
Is the maxDataPointsLimit on total data points across series?

Reply:
yeah across all unique series. Per series limits should be defined either in your proxy or in your TimeSeriesLogicalPlanner, since that's basically defined by TimeBuckets.length.